automod/rules/replies.go

```diff
@@ -78,6 +78,39 @@ func IdenticalReplyPostRule(c *automod.RecordContext, post *appbsky.FeedPost) er
 	return nil
 }
 
+// Similar to above rule but only counts replies to the same post. More aggressively applies a spam label to new accounts that are less than a day old.
+var identicalReplySameParentLimit = 3
+var identicalReplySameParentSpamLabel = 24 * time.Hour
+var _ automod.PostRuleFunc = IdenticalReplyPostSameParentRule
+
+func IdenticalReplyPostSameParentRule(c *automod.RecordContext, post *appbsky.FeedPost) error {
+	if post.Reply == nil || IsSelfThread(c, post) {
+		return nil
+	}
+
+	if ParentOrRootIsFollower(c, post) {
+		return nil
+	}
+
+	period := countstore.PeriodDay
+	bucket := c.Account.Identity.DID.String() + "/" + post.Reply.Parent.Uri + "/" + HashOfString(post.Text)
+	c.IncrementPeriod("reply-text-same-post", bucket, period)
+
+	count := c.GetCount("reply-text-same-post", bucket, period)
+	if count >= identicalReplySameParentLimit {
+		// if account is young, add spam label. otherwise just report.
+		createdAt := c.Account.CreatedAt
```

Comment:
It is tempting to find more filters to skip earlier, before the counter is even incremented. As a perf thing. We mostly have "new influx" but will get "new reactivation" as well. We could filter on "older than a day, and more than 50 posts already" just skips? I think we have the post count pre-cached on the AccountMetadata.

Maybe overthinking though, redis is fast.

Reply:
yea tbh think this makes sense in this context. aged acct more likely to be used for something else than this. maybe.